homeassistant/components/overkiz/config_flow.py

```diff
@@ -0,0 +1,113 @@
+"""Config flow for Overkiz (by Somfy) integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from aiohttp import ClientError
+from pyoverkiz.client import OverkizClient
+from pyoverkiz.const import SUPPORTED_SERVERS
+from pyoverkiz.exceptions import (
+    BadCredentialsException,
+    MaintenanceException,
+    TooManyRequestsException,
+)
+from pyoverkiz.models import obfuscate_id
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import dhcp
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import AbortFlow, FlowResult
+from homeassistant.helpers import device_registry as dr
+
+from .const import CONF_HUB, DEFAULT_HUB
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Overkiz (by Somfy)."""
+
+    VERSION = 1
+
+    async def async_validate_input(self, user_input: dict[str, Any]) -> FlowResult:
+        """Validate user credentials."""
+        username = user_input[CONF_USERNAME]
+        password = user_input[CONF_PASSWORD]
+        server = SUPPORTED_SERVERS[user_input[CONF_HUB]]
+
+        async with OverkizClient(
+            username=username, password=password, server=server
+        ) as client:
+            await client.login()
+
+            # Set first gateway as unique id
+            gateways = await client.get_gateways()
+            if gateways:
+                gateway_id = gateways[0].id
+                await self.async_set_unique_id(gateway_id)
+
+            self._abort_if_unique_id_configured()
+
+            return self.async_create_entry(title=username, data=user_input)
```

Comment:
Move this out of this function and have it return the unique id instead so that the entry isn't created in the try block

Reply:
8819bcf

Reply:
Not sure what I miss now in my test coverage, since I don't think that much changed.